be/src/exec/schema_scanner/schema_pipe_files.cpp

```diff
@@ -26,47 +27,86 @@ SchemaScanner::ColumnDesc SchemaTablePipeFiles::_s_columns[] = {
         {"DATABASE_NAME", TYPE_VARCHAR, sizeof(StringValue), false},
         {"PIPE_ID", TYPE_BIGINT, sizeof(int64_t), false},
         {"PIPE_NAME", TYPE_VARCHAR, sizeof(StringValue), false},
+
         {"FILE_NAME", TYPE_VARCHAR, sizeof(StringValue), false},
-        {"ROW_COUNT", TYPE_BIGINT, sizeof(int64_t), false},
+        {"FILE_VERSION", TYPE_VARCHAR, sizeof(StringValue), false},
+        {"FILE_ROWS", TYPE_BIGINT, sizeof(int64_t), false},
         {"FILE_SIZE", TYPE_BIGINT, sizeof(int64_t), false},
+        {"LAST_MODIFIED", TYPE_VARCHAR, sizeof(StringValue), false},
+
         {"LOAD_STATE", TYPE_VARCHAR, sizeof(StringValue), false},
+        {"STAGED_TIME", TYPE_VARCHAR, sizeof(StringValue), false},
+        {"START_LOAD_TIME", TYPE_VARCHAR, sizeof(StringValue), false},
+        {"FINISH_LOAD_TIME", TYPE_VARCHAR, sizeof(StringValue), false},
+
         {"ERROR_MSG", TYPE_VARCHAR, sizeof(StringValue), false},
-        {"ERROR_LINE_NUMBER", TYPE_BIGINT, sizeof(int64_t), false},
-        {"ERROR_COLUMN", TYPE_VARCHAR, sizeof(StringValue), false},
+        {"ERROR_COUNT", TYPE_BIGINT, sizeof(int64_t), false},
+        {"ERROR_LINE", TYPE_BIGINT, sizeof(int64_t), false},
 };
 
 SchemaTablePipeFiles::SchemaTablePipeFiles()
-    :SchemaScanner(_s_columns, sizeof(_s_columns) / sizeof(SchemaScanner::ColumnDesc)) ){
-}
+        : SchemaScanner(_s_columns, sizeof(_s_columns) / sizeof(SchemaScanner::ColumnDesc)) {}
 
 Status SchemaTablePipeFiles::start(RuntimeState* state) {
-    RETURN_IF_ERROR(SchemaScanner::start(state));
+    return SchemaScanner::start(state);
+}
 
-    return {};
+Status SchemaTablePipeFiles::_list_pipe_files() {
+    RETURN_IF(_param->ip == nullptr || _param->port == 0, Status::InternalError("unknown frontend address"));
+
+    TListPipeFilesParams params;
+    if (_param->current_user_ident) {
+        params.__set_user_ident(*_param->current_user_ident);
+    }
+    return SchemaHelper::list_pipe_files(*(_param->ip), _param->port, params, &_pipe_files_result);
 }
 
 Status SchemaTablePipeFiles::get_next(ChunkPtr* chunk, bool* eos) {
-    if (_cur_idx > 0) {
-        *eos = true;
-        return {};
+    while (_cur_row >= _pipe_files_result.pipe_files.size()) {
+        if (!_fetched) {
+            // send RPC
+            _fetched = true;
+            RETURN_IF_ERROR(_list_pipe_files());
+        } else {
+            *eos = true;
+            return Status::OK();
+        }
     }
-    fill_chunk(chunk);
     *eos = false;
-    return {};
+    return _fill_chunk(chunk);
+}
+
+DatumArray SchemaTablePipeFiles::_build_row() {
+    auto& pipe_file = _pipe_files_result.pipe_files.at(_cur_row);
+    _cur_row++;
+    return {
+            Slice(pipe_file.database_name),
```

Comment:
Slice excapes  outside, is it safe?  if _pipeline_files_result is resized or cleared?

Reply:
it's safe until now, since the RPC result is immutable after fired, and the `Scanner` only iterate the result row-wise.